src/service/core/data/query.py

```
@@ -292,7 +292,7 @@ def string_number_list(self, children):
 
     def user_expression(self, children):
         command = 'dataset.id in (SELECT id in dataset_version ' +\
-                  f'WHERE {" ".join([child.cmd for child in children])})'
+                  f'WHERE {' '.join([child.cmd for child in children])})'
         params: List = sum((child.params for child in children), [])
```

Ruff (0.15.10)

[warning] Avoid quadratic list summation

Replace with `functools.reduce`

(RUF017)